helm_tests/webserver/test_rpc_server.py

```diff
@@ -0,0 +1,923 @@
+        )
+        assert {"name": "Host", "value": "example.com"} in jmespath.search(
+            "spec.template.spec.containers[0].startupProbe.httpGet.httpHeaders", docs[0]
+        )
+
+    def test_should_add_path_to_liveness_and_readiness_and_startup_probes(self):
+        docs = render_chart(
+            values={"rpcServer": {"baseUrl": "https://example.com:21222/mypath/path"}},
+            show_only=["templates/rpc-server/rpc-server-deployment.yaml"],
+        )
+
+        assert (
+            jmespath.search("spec.template.spec.containers[0].livenessProbe.httpGet.path", docs[0])
+            == "/mypath/path/internal_api/v1/health"
+        )
+        assert (
+            jmespath.search("spec.template.spec.containers[0].readinessProbe.httpGet.path", docs[0])
+            == "/mypath/path/internal_api/v1/health"
+        )
+        assert (
+            jmespath.search("spec.template.spec.containers[0].startupProbe.httpGet.path", docs[0])
+            == "/mypath/path/internal_api/v1/health"
+        )
+
+    @pytest.mark.parametrize(
+        "revision_history_limit, global_revision_history_limit",
+        [(8, 10), (10, 8), (8, None), (None, 10), (None, None)],
+    )
+    def test_revision_history_limit(self, revision_history_limit, global_revision_history_limit):
+        values = {"rpcServer": {}}
+        if revision_history_limit:
+            values["rpcServer"]["revisionHistoryLimit"] = revision_history_limit
+        if global_revision_history_limit:
+            values["revisionHistoryLimit"] = global_revision_history_limit
+        docs = render_chart(
+            values=values,
+            show_only=["templates/rpc-server/rpc-server-deployment.yaml"],
+        )
+        expected_result = revision_history_limit if revision_history_limit else global_revision_history_limit
+        assert jmespath.search("spec.revisionHistoryLimit", docs[0]) == expected_result
+
+    @pytest.mark.parametrize("values", [{"config": {"rpcServer": {"base_url": ""}}}, {}])
+    def test_should_not_contain_host_header(self, values):
+        print(values)
+        docs = render_chart(values=values, show_only=["templates/rpc-server/rpc-server-deployment.yaml"])
+
+        assert (
+            jmespath.search("spec.template.spec.containers[0].livenessProbe.httpGet.httpHeaders", docs[0])
+            is None
+        )
+        assert (
+            jmespath.search("spec.template.spec.containers[0].readinessProbe.httpGet.httpHeaders", docs[0])
+            is None
+        )
+        assert (
+            jmespath.search("spec.template.spec.containers[0].startupProbe.httpGet.httpHeaders", docs[0])
+            is None
+        )
+
+    def test_should_use_templated_base_url_for_probes(self):
+        docs = render_chart(
+            values={
+                "rpcServer": {
+                    "baseUrl": "https://{{ .Release.Name }}.com:21222/mypath/{{ .Release.Name }}/path"
```

Comment:
hmm, do we need a base url for this component?

Reply:
with webserver it is an airflow config item, which is why the webserver deployment can ref the config value, and it doesn't need to be an item in the component.  but with rpc server, a value is needed for the same reasons (look at the deployment manifest) but, in contrast with webserver, for _rpc server_, there_ is _no_ need for an _airflow_ config for this value. that's why i added it as a component param.  WDYT?

Reply:
Forgive me for not being up to speed on 44. If there isn't a config for it, how do the other components know where to send their requests?

Reply:
you're right jed. i think i must have been confused because the config name is `core.internal_api_url` compared with `webserver.baseUrl`

Reply:
and... corrected